### PR TITLE
Make empty_symint support dynamism.

### PR DESCRIPTION
**Pull request by vanbasten23**
To address comment https://github.com/pytorch/xla/pull/4524#issuecomment-1412542586.

### CODE REVIEW

File: test/cpp/test_aten_xla_tensor.cpp [hunk 1]
```diff
@@ -5173,7 +5173,12 @@ TEST_F(AtenXlaTensorTest, TestOneIndexTransfer) {
   }
 }
 
-// Temporarily disable test. See  https://github.com/pytorch/xla/issues/4501
+// Temporarily disable test. Original issue
```
Comment (alanwaketan) [inline test/cpp/test_aten_xla_tensor.cpp:5176]: I guess this comment is not relevant to the master?

File: test/test_dynamic_shapes.py [hunk 2]
```diff
@@ -4,6 +4,7 @@
 import torch, torch_xla
 import torch_xla.core.xla_model as xm
 
+pd = torch._C._EnablePythonDispatcher()
```
Comment (alanwaketan) [inline test/test_dynamic_shapes.py:7]: Do we want to enable python dispatcher for master?

File: test/cpp/test_aten_xla_tensor.cpp (hunk 1 above)
Comment (vanbasten23) [reply test/cpp/test_aten_xla_tensor.cpp:5176]: This test is also disabled in the master branch https://github.com/pytorch/xla/blob/889a10390e388e29d804e23c99b566238c884740/test/cpp/test_aten_xla_tensor.cpp#L5176.

The comment also applies as both branches are impacted by the `nonzero` regression.

File: test/test_dynamic_shapes.py (hunk 2 above)
Comment (vanbasten23) [reply test/test_dynamic_shapes.py:7]: Please correct me if I'm wrong. To my understanding, for dynamic shape feature to run, we enable python dispatch by default.

**Review by alanwaketan — APPROVED** [review]
LGTM.

**Pull request merged**